### PR TITLE
[ExportVerilog] Legalize keywords by per-basename uniqification

### PR DESCRIPTION
**Pull request by uenoku**
This PR improves `NameCollisionResolver::getLegalName` to legalize keywords using uniqification id associated with the keywords. Previously names were legalized by a global id which was shared across a module but that implementation creates strange names.

### CODE REVIEW

File: lib/Conversion/ExportVerilog/ExportVerilogInternals.h [hunk 1]
```diff
@@ -102,7 +102,7 @@ struct NameCollisionResolver {
   llvm::StringSet<> usedNames;
 
   /// Numeric suffix used as uniquification agent when resolving conflicts.
-  size_t nextGeneratedNameID = 0;
+  llvm::StringMap<size_t> nextGeneratedNameID;
```
Comment (seldridge) [inline lib/Conversion/ExportVerilog/ExportVerilogInternals.h:105]: Using a map of Attribute to size_t may be cheaper if we are know to always use String Attr.

**Review by seldridge — APPROVED** [review]
Looks good. One thought about using a possibly simpler key.

**Review by dtzSiFive — COMMENTED** [review]
Looks functionally okay-- this should produce names more consistently (less interference from other renaming by this resolver instance), which I believe is the intention, right?

File: lib/Conversion/ExportVerilog/ExportVerilogInternals.h (hunk 1 above)
Comment (uenoku) [reply lib/Conversion/ExportVerilog/ExportVerilogInternals.h]: It looks replacing Attribute with size_t here would require more broader refactoring so i'd like to keep using strings as key within this PR.

**Pull request merged**